Framework/Algorithms/src/GeneratePythonScript.cpp

```diff
@@ -65,8 +65,8 @@ void GeneratePythonScript::init() {
       boost::make_shared<StringListValidator>(saveVersions),
       "When to specify which algorithm version was used by Mantid.");
 
-  declareProperty("IgnoreGroups", false,
-                  "Whether or not groups need to be filtered for",
+  declareProperty("ProjectRecovery", false,
```

Comment:
Instead of a flag to indicate the usage mode how about having a string list property that accepts a list of algorithms to ignore. These can then be defined by project recovery and we can leave the low-level `GeneratePythonScript` a bit more generic.

Also, does `GenerateIPythonNotebook` need a similar set of changes?

Reply:
That's a good idea let me work on that

Reply:
However I have not looked at GenerateIPythonNotebook I don't know what it does.

Reply:
It's quite similar to `GeneratePythonScript` but uses the history to create an IPython notebook rather than a plain Python script. I think there is some unfortunate code duplication between those two algorithms.